chat/client.py

```diff
@@ -0,0 +1,63 @@
+import socket
+import sys
+from threading import Thread
+
+HOST = '127.0.0.1'
+PORT = 0
+USERNAME = ''
+BUFSIZE = 1024
+
+# show received data
+def receive():
+    while (True):
+        try:
+            msg = client_socket.recv(BUFSIZE).decode('utf-8')
+            print(msg)
+
+            # client disconnected manually
+            if msg == "{quit}":
```

Comment:
changes for {quit} made on server side: lines 95-102
Did you check that other clients got next message:
`%s has left the chat.`